userspace/libscap/examples/01-open/test.c

```diff
@@ -461,36 +401,212 @@ void print_kernel_simple_consumer_syscalls()
 		}
 	}
 
-	print_ordered_syscalls(str, interesting_syscall);
+	print_sorted_syscalls(str, interesting_syscall);
 }
 
-int main(int argc, char** argv)
+/* Print all supported syscall according to the scap source you have chosen:
+ * - `KERNEL_MODULE` or `BPF_PROBE`, print all syscalls supported by these
+ *   sources. Please note: you can also print only simple consumer syscalls
+ *   by passing the CLI option `--simple_consumer`
+ * - `MODERN_BPF_PROBE`, print all syscalls that will be supported by
+ *   modern BPF probe. These syscall are a subset of actual simple consumer.
+ *   `--simple_consumer`  option will have no effect on this print.
+ * - `SCAP_FILE` not supported by this function.
+ */
+void print_supported_syscalls()
 {
-	char error[SCAP_LASTERR_SIZE];
-	int32_t res;
-	scap_evt* ev;
-	uint16_t cpuid;
+	switch(source)
+	{
+	case KERNEL_MODULE:
+	case BPF_PROBE:
+		print_actual_drivers_syscalls();
+		break;
 
-	if(signal(SIGINT, signal_callback) == SIG_ERR)
+	case MODERN_BPF_PROBE:
+		print_modern_probe_syscalls();
+		break;
+
+	case SCAP_FILE:
+	default:
+		printf("Scap source not supported! Bye!");
+	}
+}
+
+/*=============================== PRINT SUPPORTED SYSCALLS ===========================*/
+
+/*=============================== PRINT SYSCALLS VALIDATION ===========================*/
+
+void print_validation()
```

Comment:
I'd rather call this just `validate_syscalls` :D

Reply:
if you take a look at the menu there are 3 categories of options:
```
----------------------- MENU -----------------------
------> SCAP SOURCES
'--kmod': enable the kernel module.
'--bpf <probe_path>': enable the BPF probe.
'--modern_bpf': enable modern BPF probe.
'--scap_file <file.scap>': read events from scap file.

------> CONFIGURATIONS
'--simple_consumer': enable the simple consumer mode. (default: disabled)
'--num_events <num_events>': number of events to catch before terminating. (default: UINT64_MAX)
'--evt_type <event_type>': every event of this type will be printed to console. (default: -1, no print)

------> PRINT OPTIONS
'--print_syscalls': print all supported syscalls with different sources and configurations.
'--print_validation': print some validation checks.
'--help': print this menu.
-----------------------------------------------------
```

If we change the name to `print_validation` we will have to add another category, something like `validation options`, do you agree with this choice?

Reply:
This is the new menu 
```
----------------------- MENU -----------------------
------> SCAP SOURCES
'--kmod': enable the kernel module.
'--bpf <probe_path>': enable the BPF probe.
'--modern_bpf': enable modern BPF probe.
'--scap_file <file.scap>': read events from scap file.

------> CONFIGURATIONS OPTIONS
'--simple_consumer': enable the simple consumer mode. (default: disabled)
'--num_events <num_events>': number of events to catch before terminating. (default: UINT64_MAX)
'--evt_type <event_type>': every event of this type will be printed to console. (default: -1, no print)

------> VALIDATION OPTIONS
'--validate_syscalls': validation checks.

------> PRINT OPTIONS
'--print_syscalls': print all supported syscalls with different sources and configurations.
'--help': print this menu.
-----------------------------------------------------
```